File: gcj2010QRB.hs
#!/usr/bin/env runghc
import Control.Monad
import Data.List

main = do
  c <- liftM lines getContents
  let ss = zipWith (\i r->"Case #"++show i++": "++r) [1..] $ solve c
  mapM_ putStrLn ss

solve (c:cs) = map u $ analyze cs
  where
    u a = show $ (g - x `mod` g) `mod` g
      where
        (x:xs) = sort $ nub a
        g = (foldl1 gcd . map (subtract x)) xs
    analyze = map (tail . map read . words)

File: gcj2010QRB.py
#!/usr/bin/env python
import sys
import fractions

def solve():
    t = sorted(set(map(long,sys.stdin.readline().split()[1:])))
    g = reduce(fractions.gcd,map(lambda x: x-t[0],t[1:]))
    return (g-t[0]%g)%g

for x in range(int(sys.stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))
